nengo/dists.py

```diff
@@ -306,6 +346,358 @@ def sample(self, n, d=None, rng=np.random):
         return samples
 
 
+class QuasirandomSequence(Distribution):
+    """Sequence for quasi Monte Carlo sampling the ``[0, 1]``-cube.
+
+    This is similar to ``np.random.uniform(0, 1, size=(num, d))``, but with the
+    additional property that each ``d``-dimensional point is uniformly scattered.
+
+    This is based on the tutorial and code from [#]_.
+
+    See Also
+    --------
+    ScatteredHypersphere
+
+    References
+    ----------
+    .. [#] Martin Roberts. "The Unreasonable Effectiveness of Quasirandom Sequences."
+       http://extremelearning.com.au/unreasonable-effectiveness-of-quasirandom-sequences/
+
+    Examples
+    --------
+
+    .. testcode::
+
+       from nengo.dists import QuasirandomSequence
+       rd = QuasirandomSequence().sample(10000, 2)
+
+       plt.scatter(*rd.T, c=np.arange(len(rd)), cmap='Blues', s=7)
+    """
+
+    @staticmethod
+    def _phi(d, tol=1e-10):
+        """Newton-Raphson-Method to calculate ``g = phi_d``."""
+        x = 1.0
+        for _ in range(100):
+            dx = (x ** (d + 1) - x - 1) / ((d + 1) * x ** d - 1)
+            x -= dx
+            if abs(dx) < tol:
+                break
+        else:
+            raise ConvergenceError("'phi' computation did not converge for d=%s" % d)
+
+        return x
+
+    def sample(self, n, d=1, rng=np.random):
+        if d is None or d < 1:  # check this, since other dists allow d = None
+            raise ValidationError("Dimensions must be a positive integer", "d")
+
+        seed = rng.uniform(0, 1, size=d)
+        if d == 1:
+            # Tile the points optimally
+            z = (seed + (1 / n) * np.arange(n, dtype=np.float64)) % 1
+            return z[:, np.newaxis]
+        else:
+            phi_inv = 1.0 / self._phi(d)
+            alpha = phi_inv ** np.arange(1, d + 1)
+            z = (seed + np.outer(np.arange(1, n + 1), alpha)) % 1
+            return z
+
+
+class _betaincinv22:
```

Comment:
I might move this to the bottom so it's not distracting from the actual distributions.